src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs

```diff
@@ -209,18 +209,25 @@ async fn test_append_prev_block_sequence_integrity_broken() {
 
     let hash = chain
         .append(
-            MetadataFactory::metadata_block(MetadataFactory::add_data().interval(0, 9).build())
-                .prev(&hash, 1)
-                .build(),
+            MetadataFactory::metadata_block(
+                MetadataFactory::add_data()
+                    .new_offset_interval(0, 9)
+                    .build(),
+            )
+            .prev(&hash, 1)
+            .build(),
             AppendOpts::default(),
         )
         .await
         .unwrap();
 
-    let block_too_low =
-        MetadataFactory::metadata_block(MetadataFactory::add_data().interval(10, 19).build())
-            .prev(&hash, 1 /* should be 2 */)
-            .build();
+    let block_too_low = MetadataFactory::metadata_block(
+        MetadataFactory::add_data()
+            .new_offset_interval(10, 19)
+            .build(),
+    )
+    .prev(&hash, 1 /* should be 2 */)
```

Comment:
please add a comment w/ context why it should be

Reply:
I think the code here is fairly self-documenting:
- the test name is about block sequence integrity
- right below we assert on sequence integrity error